File: openmole-ui-ide/openmole-ide-ant/org.openmole.ui.ide/src/org/openmole/ui/ide/workflow/implementation/IContainerUI.java
/*
 *  Copyright (C) 2011 Mathieu Leclaire <mathieu.leclaire@openmole.org>
 * 
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 * 
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 * 
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package org.openmole.ui.ide.workflow.implementation;

import java.util.Collection;
import org.openmole.commons.exception.UserBadDataError;

/**
 *
 * @author Mathieu Leclaire <mathieu.leclaire@openmole.org>
 */
public interface IContainerUI {
    void register(IEntityUI entity);
    void removeEntity(IEntityUI entity) throws UserBadDataError;
    IEntityUI get(String st,Class type) throws UserBadDataError;
    Collection<IEntityUI> getAll();
    void clearAll();
    boolean contains(IEntityUI entity);
}

File: gui/src/main/scala/org/openmole/ui/ide/workflow/model/IContainerUI.scala
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package org.openmole.ui.ide.workflow.model

trait IContainerUI {
  def register(entityUI: IEntityUI)

  def removeEntity(entityUI: IEntityUI)
  
  def get(st: String, entityType: Class[_]): IEntityUI
  
  def getAll: Collection[IEntityUI]
  
  def clearAll
  
  def contains(entityUI: IEntityUI)
}


//public interface IContainerUI {
//    void register(IEntityUI entity);
//    void removeEntity(IEntityUI entity) throws UserBadDataError;
//    IEntityUI get(String st,Class type) throws UserBadDataError;
//    Collection<IEntityUI> getAll();
//    void clearAll();
//    boolean contains(IEntityUI entity);
//}